common/djangoapps/student/helpers.py

```diff
@@ -270,6 +261,49 @@ def get_next_url_for_login_page(request):
     return redirect_to
 
 
+def get_redirect_to(request):
+    """
+    Determine the redirect url and return if safe
+    :argument
+        request: request object
+
+    :returns: redirect url if safe else None
+    """
+    redirect_to = request.GET.get('next')
+    header_accept = request.META.get('HTTP_ACCEPT', '')
+
+    # If we get a redirect parameter, make sure it's safe i.e. not redirecting outside our domain.
+    # Also make sure that it is not redirecting to a static asset and redirected page is web page
+    # not a static file. As allowing assets to be pointed to by "next" allows 3rd party sites to
+    # get information about a user on edx.org. In any such case drop the parameter.
+    if redirect_to:
+        if not http.is_safe_url(redirect_to):
+            log.warning(
+                u'Unsafe redirect parameter detected after login page: %(redirect_to)r',
+                {"redirect_to": redirect_to}
+            )
+            redirect_to = None
+        else:
+            if 'text/html' not in header_accept:
+                log.warning(
+                    u'Redirect to non html content detected after login page: %(redirect_to)r',
+                    {"redirect_to": redirect_to}
+                )
+                redirect_to = None
+            else:
+                mime_types = mimetypes.guess_type(redirect_to, strict=False)
+                for mime_type in mime_types:
+                    if mime_type and "image" in mime_type:
```

Comment:
Agreed. But just as we verified that checking the header wouldn't cause unintended false negatives, we should do the same for the mimetype solution, that is, we need to have confidence that this won't accidentally break anything.